lib/sanctify/matcher.rb

```diff
@@ -0,0 +1,27 @@
+module Sanctify
+  class Matcher
+    attr_reader :description, :regex, :config
+    def initialize(description, regex, config)
```

Comment:
It seems a little odd for an individual matcher to have awareness of the whole gem config (and also expose it via an attribute reader). Considering that all we're using this config for is to discern whether the matcher is disabled, it'd be a tad cleaner to just inject a `disabled:` attribute here.

A usability note: when a method's parameters aren't inherently obvious (almost always true when a method has more than one parameter), it can help to use named parameters instead of positional ones.

```ruby
# A. Positional with options
Sanctify::Matcher.new('Hello, world!', /^Hello, world!$/, disabled: true)

# B. All named parameters
Sanctify::Matcher.new(description: 'Hello, world!', regex: /^Hello, world!$/, disabled: true)
```

🚲 🏚  In comparing the two here, option A seems to make sense and is basically what you have here, with the only exception being that the last argument is an options hash instead of the global'ish gem config.